.circleci/config.yml

```diff
@@ -255,6 +255,7 @@ jobs:
             export GIT_DATE="$(git show -s --format='%ct')"
             export GIT_VERSION="<<pipeline.git.tag>>"
             export PLATFORMS="<<parameters.platforms>>"
+            export VERSION="<<pipeline.git.tag>>"
```

Comment:
I think overall, this approach should work.

I have two issues with it though.

1. Won't we end up with output like:

```
# op-node --version
op-node version op-node/v1.2.1-deadbeef-deadbeef
```

(ie, where the component name is a part of the version string)?

2. This 'VERSION' env variable is going to override the version of all built components.  So long as the CI is structured as it is, this should be fine, as only one 'tag' is ever built.  But, it still worries me that we're ending up with a build cache full of incorrectly tagged binaries.  I think it's probably worth taking something upstream so that we don't have to risk things getting subtly out of sync.  

I think (1) is probably worth fixing before merging.  We can deal with (2) later.

Reply:
This is our image output
```
op-node-1  | op-node version v0.0.0-068cf246-1706132859
```
According to the code,
```
var VersionWithMeta = opservice.FormatVersion(version.Version, GitCommit, GitDate, version.Meta)
```
The `version` is v0.0.0 when we build the image.

Reply:
Another way to fix this is to update
```
LDFLAGSSTRING +=-X github.com/ethereum-optimism/optimism/op-node/version.Version=$(VERSION)
```
to 
```
LDFLAGSSTRING +=-X github.com/ethereum-optimism/optimism/op-node/version.Version=$(GIT_VERSION)
```
but this doesn't make sense to me.

Reply:
Right, I'd agree that setting `VERSION` is the way it was designed to work.  But, `pipeline.git.tag` is going to be `op-node/v1.2.1` and what I think you want is just `v1.2.1`.  I was just suggesting that you probably want to strip everything out of the git tag up to and including the first `/`.